lib/config.js

```diff
@@ -113,12 +118,45 @@ function fromKubeconfig(kubeconfig, current) {
 
 module.exports.fromKubeconfig = fromKubeconfig;
 
+function normalizePathFn(cfgPath) {
+  const configDir = path.dirname(cfgPath);
+  return (subPath) => {
+    const normalizedPath = path.normalize(subPath);
+    return path.isAbsolute(normalizedPath) ? normalizedPath : path.join(configDir, normalizedPath);
+  };
+}
+
+function mapSpecificKeysFn(mapFn, srcKey) {
+  return (source) => {
+    return Object.keys(source).reduce((dest, key) => {
+      dest[key] = (key === srcKey) ? mapFn(source[key]) : source[key];
+      return dest;
+    }, {});
+  };
+}
+
+function mapCertificates(cfgPath, config) {
```

Comment:
I appreciate the work to implement the helpers / map functions, but it took me a while to understand it completely.  I'd worry it might give others who wish to extend or maintain it pause also.  What about a simple-stupid implementation?  E.g.,

```js
config.clusters.filter(cluster => cluster.cluster['certificate-authority']).forEach(cluster => {
  cluster.cluster['certificate-authoriy'] = path.resolve(cfgPath, cluster.cluster['certificate-authority']);
});

config.users.filter(user => user.user['client-certificate']).forEach(user => {
  user.user['client-certificate'] = path.resolve(cfgPath, user.user['client-certificate']);
});

config.users.filter(user => user.user['client-key']).forEach(user => {
  user.user['client-key'] = path.resolve(cfgPath, user.user['client-key']);
});
```

Reply:
Well, that is why I was asking which solution you prefer. This was the immutable solution, the mutable looked something similar to your approach without the duplication.

Reply:
I did not know `path.resolve()`. Thanks!